File: Domain/Entities/Order.cs
using Domain.Common;
using Domain.Enum;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{

        public class Order : BaseAuditableEntity
        {
        public OrderType OrderType { get; set; } = OrderType.Purchase;
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Ready;
        public int CustomerId { get; set; }
        public Customer Customer { get; set; }
        public int EngineerId { get; set; }
        public Engineer Engineer { get; set; }
        public byte[]? Document { get; set; }
        public List<Item>? OrderItems { get; set; }
        public List<Part>? OrderParts { get; set; }
    }
}

File: API/client-app/src/app/shared/models/order.ts
import { ICustomer } from "./customer"
import { IEngineer } from "./engineer"
import { IItem } from "./item"
import { IPart } from "./part"

export interface IOrder {
  id: number
  orderType: string
  orderStatus: string
  customer: string
  engineer: string
}

export interface IPartNumber {
  qty: number
  partNumber: string
}



export interface IOrderDetail {
  orderType: number
  orderStatus: number
  customerId: number
  customer: ICustomer
  engineerId: number
  engineer: IEngineer
  document: any
  orderItems: IItem[]
  orderParts: IPart[]
  created: string
  createdBy: any
  lastModified: string
  lastModifiedBy: any
  id: number
}
